ibm/service/power/resource_ibm_pi_instance_action.go

```diff
@@ -0,0 +1,213 @@
+// Copyright IBM Corp. 2017, 2021 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package power
+
+import (
+	"context"
+	"fmt"
+
+	st "github.com/IBM-Cloud/power-go-client/clients/instance"
+
+	"github.com/IBM-Cloud/power-go-client/power/models"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/validate"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"log"
+	"time"
+)
+
+func ResourceIBMPIInstanceAction() *schema.Resource {
+	return &schema.Resource{
+
+		CreateContext: resourceIBMPIInstanceActionCreate,
+		ReadContext:   resourceIBMPIInstanceActionRead,
+		UpdateContext: resourceIBMPIInstanceActionUpdate,
+		DeleteContext: resourceIBMPIInstanceActionDelete,
+		Importer:      &schema.ResourceImporter{},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(15 * time.Minute),
+			Update: schema.DefaultTimeout(15 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			// Arguments
+			Arg_CloudInstanceID: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "PI Cloud instnce id",
+			},
+			Arg_PVMInstanceId: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "PVM Instance ID",
+			},
+			Arg_PVMInstanceActionType: {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.ValidateAllowedStringValues([]string{"start", "stop", "hard-reboot", "soft-reboot", "immediate-shutdown"}),
```

Comment:
The swagger defines another action: reset-state

Reply:
Do you know what is the resulting status of the instance after applying that state?

Reply:
FYI: after reset-state instance status is ACTIVE and health status is CRITICAL

Reply:
Moises said It should go back to active. The reset is only used for VMs that are still running but it failed a DLPAR operation and went to Error state

Reply:
Yes. I have made the changes. Testing it was very difficult to find an instance with ERROR state to start with, hence making it a new test case.